### DISCUSSION TITLE
ref.current.selection bugs

### DISCUSSION THREAD

**Discussion by Alkimay** [nick-keller/react-datasheet-grid#139] [Q&A] [answered]
hey, so this object is being updated only when I press another row different from the current one.
for example, if i am pressing cel 1,2 and then 1,4  its not updating, if i press 1,2 and then 2,* it is updating,

it seems like an easy fix could u please fix it?>

**Comment by nick-keller** [comment]
Thx I will look into it!

**Comment by Alkimay** [comment]
> Thx I will look into it!

also, if u update the data from outside(by code or function) some row and on mouse/marker is on this row/cell its not rendering the value (it is changing the data), only when switching to other row its updates and renders the update

**Comment by nick-keller** [comment]
This is done on purpose, the text column uses an uncontrolled component while editing for performance. If you need a specific behavior you can create your own component.

**Comment by nick-keller** [comment, marked as answer]
Hello @Alkimay ,
I cannot reproduce this error, I assume it comes from your code. Here is what I did:

```tsx
  const ref = useRef<DataSheetGridRef>(null)

  useEffect(() => {
    // Log selection every second
    setInterval(() => {
      console.log(ref.current?.selection)
    }, 1000)
  }, [])

  return <DataSheetGrid ref={ref} ... />
```

Can you please provide code to showcase the issue?
